x-pack/platform/plugins/private/upgrade_assistant/server/lib/es_deprecations_status/get_corrective_actions.ts

```diff
@@ -62,9 +56,9 @@ export const getCorrectiveAction = (deprecation: BaseDeprecation): CorrectiveAct
   const clusterSettingDeprecation = metadata?.actions?.find(
     (action) => action.action_type === 'remove_settings' && typeof index === 'undefined'
   );
-  const requiresReindexAction = ES_INDEX_MESSAGES_REQUIRING_REINDEX.some((regexp) =>
-    regexp.test(message)
-  );
+  const requiresReindexAction =
+    (type === 'index_settings' || type === 'node_settings') &&
+    (deprecation.metadata as IndexActionMetadata)?.reindex_required === true;
```

Comment:
I am lacking knowledge for the `node_settings` deprecations, but I've verified and the meta property is present for the `index_settings` ones.